src/main/java/org/folio/bulkops/service/SrsService.java

```diff
@@ -61,6 +61,7 @@ public void retrieveMarcInstancesFromSrs(List<String> instanceIds, BulkOperation
     } else {
       bulkOperation.setLinkToCommittedRecordsMarcFile(null);
     }
+    bulkOperationRepository.save(bulkOperation);
```

Comment:
Why do we need extra saving of bulkOperation here? The next method private void completeBulkOperation(BulkOperation bulkOperation, int committedNumOfRecords) also saves bulkOperation. linkToCommittedRecordsMarcFile can be stored in bulkOperation in completeBulkOperation method.

Reply:
Method completeBulkOperation stores bulkOperation only after retrieving it by id and case bulkOperation.setLinkToCommittedRecordsMarcFile(null) remains unsaved. This was the reason why we had a file in Actions to download and 500 error.